lib/Database/Database.cpp

```diff
@@ -173,8 +173,16 @@ Database::Implementation::create(StringRef path, bool readonly, Optional<size_t>
     db->DBEnv = lmdb::env::create();
     db->DBEnv.set_max_dbs(14);
 
-    // Start with 64MB. We'll update with the actual size after we open the database.
-    db->MapSize = initialDBSize.getValueOr(64ULL*1024ULL*1024ULL);
+    uint64_t dbFileSize = 0;
+    if (existingDB) {
+      if (std::error_code ec = llvm::sys::fs::file_size(dbPath + "/data.mdb", dbFileSize)) {
+        LOG_WARN_FUNC("failed reading database file size " << dbPath << "/data.mdb: " << ec.message());
+      }
+    }
+    // Start with 64MB.
+    uint64_t initialSize = initialDBSize.getValueOr(64ULL*1024ULL*1024ULL);
+
+    db->MapSize = std::max(dbFileSize, initialSize);
```

Comment:
Would there be any harm in just always using the existing size if there's a database? Ie. could we just initialize `dbFileSize` to `initialDBSize.getValueOr...` and overwrite it if there's an existing database? Assuming `file_size` doesn't set `dbFileSize` unless it succeeds that is.